lib/modules/dosomething/dosomething_user/dosomething_user_social_auth.inc

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/**
+ * Constructs a new Drupal user for the given account details to represent a Facebook user.
+ * @param  array $account_details  All of the properties to instantiate
+ * @return user
+ */
+function dosomething_user_make_facebook_user($account_details) {
+  $edit = [
+    'mail' => $account_details['email'],
+    'name' => $account_details['email'],
+    'pass' => user_password(),
+    'status' => 1,
+    'created' => REQUEST_TIME,
+  ];
+
+  dosomething_user_set_fields($edit, $account_details);
+
+  $user = user_save('', $edit);
+  user_save($user, ['name' => $user->uid]);
+
+  return $user;
+}
+
+/**
+ * Registers a new user with Northstar (or upserts with existing) & returns a Drupal account
+ * @param  array $account_details Account details for the user to register
+ * @return user
+ */
+function dosomething_user_register_facebook_user($account_details) {
```

Comment:
I can add a check sure, but isn't that reinforcing the idea of Drupal being the gate keeper of all things users? (and eventually *tm this is just gonna be a replica of northstar anyway)